geoh5py/objects/surveys/electromagnetics/airborne_tem.py

```diff
@@ -0,0 +1,388 @@
+            )
+        elif isinstance(value, uuid.UUID):
+            self.edit_metadata(
+                {"Inline offset value": None, "Inline offset property": value}
+            )
+        else:
+            raise TypeError(
+                "Input 'inline_offset' must be one of type float or uuid.UUID"
+            )
+
+    @property
+    def loop_radius(self):
+        """Transmitter loop radius"""
+        if "Loop radius" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Loop radius"]
+
+        return None
+
+    @loop_radius.setter
+    def loop_radius(self, value: float):
+        if not isinstance(value, float):
+            raise TypeError("Input 'loop_radius' must be of type 'float'")
+        self.edit_metadata({"Loop radius": value})
+
+    @property
+    def pitch(self) -> float | uuid.UUID | None:
+        """Pitch angle(s) of the transmitter coil"""
+        if "Pitch value" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Pitch value"]
+        if "Pitch property" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Pitch property"]
+        return None
+
+    @pitch.setter
+    def pitch(self, value: float | uuid.UUID):
+        if isinstance(value, float):
+            self.edit_metadata({"Pitch value": value, "Pitch property": None})
+        elif isinstance(value, uuid.UUID):
+            self.edit_metadata({"Pitch value": None, "Pitch property": value})
+        else:
+            raise TypeError("Input 'pitch' must be one of type float or uuid.UUID")
+
+    @property
+    def relative_to_bearing(self):
+        """Data relative_to_bearing"""
+        if "Angles relative to bearing" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Angles relative to bearing"]
+
+        return None
+
+    @relative_to_bearing.setter
+    def relative_to_bearing(self, value: bool):
+        if not isinstance(value, bool):
+            raise TypeError("Input 'relative_to_bearing' must be one of type 'bool'")
+        self.edit_metadata({"Angles relative to bearing": value})
+
+    @property
+    def roll(self) -> float | uuid.UUID | None:
+        """Roll angle(s) of the transmitter coil."""
+        if "Roll value" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Roll value"]
+        if "Roll property" in self.metadata["EM Dataset"]:
+            return self.metadata["EM Dataset"]["Roll property"]
+        return None
+
+    @roll.setter
+    def roll(self, value: float | uuid.UUID):
+        if isinstance(value, float):
+            self.edit_metadata({"Roll value": value, "Roll property": None})
+        elif isinstance(value, uuid.UUID):
+            self.edit_metadata({"Roll value": None, "Roll property": value})
+        else:
+            raise TypeError("Input 'roll' must be one of type float or uuid.UUID")
```

Comment:
**praise:** nice job with type hint for parameter and safe type checking at runtime ! 🥇